cpu/cc26x2_cc13x2/osc.c

```diff
@@ -0,0 +1,102 @@
+/*
+ * Copyright (C) 2020 Locha Inc
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup         cpu_cc26x2_cc13x2
+ * @{
+ *
+ * @file
+ * @brief           CC26x2/CC13x2 Oscillator functions
+ *
+ * @author          Jean Pierre Dudey <jeandudey@hotmail.com>
+ * @}
+ */
+
+#include <assert.h>
+#include "cpu.h"
+
+uint32_t osc_clock_source_get(uint32_t sclk)
+{
+    assert((sclk & OSC_SRC_CLK_LF) || (sclk & OSC_SRC_CLK_HF));
+
+    /* Return the source for the selected clock. */
+    uint32_t src;
+    if (sclk == OSC_SRC_CLK_LF) {
+        src = (DDI_0_OSC->STAT0 & DDI_0_OSC_STAT0_SCLK_LF_SRC_m) >>
+              DDI_0_OSC_STAT0_SCLK_LF_SRC_s;
+    }
+    else {
+        src = (DDI_0_OSC->STAT0 & DDI_0_OSC_STAT0_SCLK_HF_SRC_m) >>
+              DDI_0_OSC_STAT0_SCLK_HF_SRC_s;
+    }
+
+    return src;
+}
+
+void osc_clock_source_set(uint32_t sclk, uint32_t osc)
+{
+    assert((sclk & OSC_SRC_CLK_LF) || (sclk & OSC_SRC_CLK_HF));
+    assert((osc == OSC_RCOSC_HF) || (osc == OSC_RCOSC_LF) ||
+            (osc == OSC_XOSC_HF) || (osc == OSC_XOSC_LF));
+
+    if (sclk & OSC_SRC_CLK_HF) {
+        uint32_t mask = DDI_0_OSC_CTL0_SCLK_HF_SRC_SEL_m;
+        uint32_t ctl = osc << DDI_0_OSC_CTL0_SCLK_HF_SRC_SEL_s;
+
+        /* Use a 16-bit masked write, target bits are on the lower 16-bit
+         * half */
+        DDI_0_OSC_M16->CTL0.LOW = (mask << 16) | ctl;
+    }
+    else if (sclk & OSC_SRC_CLK_LF) {
+        /* Configure the low frequency source clock. */
+        uint32_t mask = DDI_0_OSC_CTL0_SCLK_LF_SRC_SEL_m;
+        uint32_t ctl = osc << DDI_0_OSC_CTL0_SCLK_LF_SRC_SEL_s;
+
+        /* Use a 16-bit masked write, target bits are on the lower 16-bit
+         * half */
+        DDI_0_OSC_M16->CTL0.LOW = (mask << 16) | ctl;
+    }
+}
+
+bool osc_hf_source_ready(void)
+{
+    if (DDI_0_OSC->STAT0 & DDI_0_OSC_STAT0_PENDINGSCLKHFSWITCHING) {
+        return true;
+    }
+    return false;
+}
+
+void osc_hf_source_switch(void)
+{
+    /* Read SCLK_HF target clock */
+    uint32_t hf_src = DDI_0_OSC->CTL0 & DDI_0_OSC_CTL0_SCLK_HF_SRC_SEL_m;
+
+    /* If target clock source is RCOSC, change clock source for DCDC to RCOSC */
+    if (hf_src == DDI_0_OSC_CTL0_SCLK_HF_SRC_SEL_RCOSC) {
+        /* Force DCDC to use RCOSC before switching SCLK_HF to RCOSC */
+        uint32_t mask = DDI_0_OSC_CTL0_CLK_DCDC_SRC_SEL_m;
+        uint32_t data = (DDI_0_OSC_CTL0_CLK_DCDC_SRC_SEL_m >> 16);
```

Comment:
```suggestion
        uint32_t data = DDI_0_OSC_CTL0_CLK_DCDC_SRC_SEL_m >> 16;
```

Makes it easier to compare with the line above.